api/types/trait/convert/v1/trait.go

```diff
@@ -33,7 +35,17 @@ func FromProto(traits []*traitv1.Trait) trait.Traits {
 // ToV1 converts a map of string to string array to an array of v1 traits.
 func ToProto(traits trait.Traits) []*traitv1.Trait {
 	out := make([]*traitv1.Trait, 0, len(traits))
-	for key, values := range traits {
+	sortedKeys := make([]string, 0, len(traits))
+
+	// Sort the keys so that the resulting order of the traits is deterministic for equivalent
+	// maps.
```

Comment:
Why does the order matter? Can't we disregard the ordering in our tests?

Reply:
The ordering doesn't really matter either way, but it means that an incoming trait map can produce two different (valid) lists of traits, so `proto.Equal` won't work reliably. I can't really ignore that in proto.Equal without forcing the sort order of the resulting message. We could potentially disregard it by forcing an explicit message rather than the testing method I'm using, but it seems reasonable for `ToProto` to produce a reliable outcome IMO.